static/js/webrtc.js

```diff
@@ -496,6 +544,20 @@ var rtc = (function() {
       window.navigator.mediaDevices
         .getUserMedia(mediaConstraints)
         .then(function(stream) {
+          // Disable audio and/or video according to user/site settings.
+          // Do this before setting `localStream` to avoid a race condition
+          // that might flash the video on for an instant before disabling it.
```

Comment:
I just thought of this. I figured that if we set `localStream` (a global var) before disabling video, and this is somehow preempted by something (ex `receiveMessage` which starts connecting `localStream` to the outside world), the other side might see a frame or something.

Still, I wonder if there's some sort of hardware or browser implementation issue that will, let's say, put a delay in the the disabling of the video.

Thoughts?

Reply:
Well, I wrote as many race condition tests as I could think of. Perhaps we should run them on a few browsers.